src/Plugin/RulesAction/EntitySaveDeriver.php

```diff
@@ -0,0 +1,96 @@
+use Drupal\Core\StringTranslation\TranslationInterface;
+use Drupal\rules\Context\ContextDefinition;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Derives entity save plugin definitions based on content entity types.
+ *
+ * @see \Drupal\rules\Plugin\RulesAction\EntitySave
+ */
+class EntitySaveDeriver extends DeriverBase implements ContainerDeriverInterface {
+  use StringTranslationTrait;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface;
+   */
+  protected $entityFieldManager;
+  /**
+   * Saves a new EntitySaveDeriver object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\StringTranslation\TranslationInterface $string_translation
+   *   The string translation service.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entity_field_manager, TranslationInterface $string_translation) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+    $this->stringTranslation = $string_translation;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, $base_plugin_id) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager'),
+      $container->get('string_translation')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    foreach ($this->entityTypeManager->getDefinitions() as $entity_type_id => $entity_type) {
+      // Only allow content entities and ignore configuration entities.
+      if (!$entity_type instanceof ContentEntityTypeInterface) {
+        continue;
+      }
+
+      $this->derivatives[$entity_type_id] = [
+        'label' => $this->t('Save @entity_type', ['@entity_type' => $entity_type->getLowercaseLabel()]),
+        'category' => $entity_type->getLabel(),
+        'entity_type_id' => $entity_type_id,
+        'context' => [
+          'entity' => ContextDefinition::create("entity:$entity_type_id")
+            ->setLabel($entity_type->getLabel())
+            ->setDescription($this->t('Specifies the entity, which should be saved permanently.'))
+            ->setRequired(TRUE),
+          'immediate' => ContextDefinition::create('boolean')
+            ->setLabel($this->t('Force saving immediately'))
+            ->setDescription($this->t('Usually saving is postponed till the end of the evaluation, so that multiple saves can be fold into one. If this set, saving is forced to happen immediately.'))
+            ->setDefaultValue(FALSE)
+            ->setRequired(FALSE),
```

Comment:
Unrelated, but this should be actually required. NULL is invalid, only TRUE or FALSE is fine.